x/qgb/abci_test.go

```diff
@@ -111,13 +111,78 @@ func TestSetDataCommitment(t *testing.T) {
 	input, ctx := testutil.SetupFiveValChain(t)
 	qk := input.QgbKeeper
 
-	input.Context = ctx.WithBlockHeight(int64(qk.GetDataCommitmentWindowParam(ctx)))
-	vs, err := qk.GetCurrentDataCommitment(ctx)
+	ctx = ctx.WithBlockHeight(int64(qk.GetDataCommitmentWindowParam(ctx)))
+	dc, err := qk.GetCurrentDataCommitment(ctx)
 	require.Nil(t, err)
-	err = qk.SetAttestationRequest(ctx, &vs)
+	err = qk.SetAttestationRequest(ctx, &dc)
 	require.Nil(t, err)
 
-	require.Equal(t, uint64(1), qk.GetLatestAttestationNonce(ctx))
+	require.Equal(t, uint64(1), qk.GetLatestAttestationNonce(input.Context))
+}
+
+func TestGetDataCommitment(t *testing.T) {
```

Comment:
Does this work? [90593fd](https://github.com/celestiaorg/celestia-app/pull/1695/commits/90593fd72a94445371786f6e49970704996e4d7c)

Reply:
This definitely helps. I wonder if it's also possible to modify the ranges to make them easier for humans to read. What if the start of the range was inclusive and the end of the range was exclusive. Then for a data commitment window of 400 that starts at `0`, we'd have:

0 - 400
400 - 800

and if we had to start at `1` because weirdness around the genesis block:

1 - 401
401 - 801

It's a bit easier to tell that these ranges are the correct size because end - start = data commitment window

Reply:
A few ideas to make this test easier to read:

1. Split the long test into multiple test cases, potentially one per range
2. Extract a range struct that specifies a begin and end block for a data commitment window. Then assert that the expected range matches the result's data commitment `BeginBlock` / `EndBlock`

Reply:
> This definitely helps. I wonder if it's also possible to modify the ranges to make them easier for humans to read. What if the start of the range was inclusive and the end of the range was exclusive. Then for a data commitment window of 400 that starts at 0, we'd have

We definitely could. I am using the inclusive ranges because when we say: `beginBlock: 1, endBlock: 100`, we automatically assume that it's inclusive. And if it's not, we would need to add a comment for it everywhere we use it. Especially if we switch the using `firstBlock`, and `lastBlock` words. Thus, I kept using the inclusive ranges.


> A few ideas to make this test easier to read:
> Split the long test into multiple test cases, potentially one per range
> Extract a range struct that specifies a begin and end block for a data commitment window. Then assert that the expected range matches the result's data commitment BeginBlock / EndBlock

We could do that but I wanted to have the sequential execution of the test and asserting that the ranges are continuous (if it's the right word to use here). That's why I opted for a sequential list of actions.

Reply:
not blocking on it, but fwiw I'd +1 any efforts to make the test more readable. Its testing complex cases, which I know isn't easy to abstract or make readable, but it is currently very beefy

perhaps we could break into test cases? it looks like we're comparing begin, endblock, and nonce 4 separate times, could we keep those heights in a test case as expected values? then we're also setting the attestation request between each iteration, could that be part of the test input?

Reply:
Is this better [eaf055c](https://github.com/celestiaorg/celestia-app/pull/1695/commits/eaf055c9b701f668ae6d66181cbbdd46326e6110) ?

Reply:
yes! thanks!